pyogrio/_err.pyx

```diff
@@ -1,25 +1,40 @@
-# ported from fiona::_err.pyx
-from enum import IntEnum
+"""Error handling code for GDAL/OGR.
+
+Ported from fiona::_err.pyx
+"""
+
+import contextlib
 import warnings
+from contextvars import ContextVar
+from enum import IntEnum
+from itertools import zip_longest
 
 from pyogrio._ogr cimport (
     CE_None, CE_Debug, CE_Warning, CE_Failure, CE_Fatal, CPLErrorReset,
     CPLGetLastErrorType, CPLGetLastErrorNo, CPLGetLastErrorMsg, OGRErr,
-    CPLErr, CPLErrorHandler, CPLDefaultErrorHandler, CPLPushErrorHandler)
+    CPLErr, CPLErrorHandler, CPLDefaultErrorHandler, CPLPopErrorHandler,
+    CPLPushErrorHandler, CPLPushErrorHandlerEx)
+
+_ERROR_STACK = ContextVar("error_stack")
+_ERROR_STACK.set([])
 
 
 # CPL Error types as an enum.
 class GDALError(IntEnum):
+    """GDAL error types/classes.
+    
+    GDAL doc: https://gdal.org/en/latest/doxygen/cpl__error_8h.html#a463ba7c7202a505416ff95b1aeefa2de
+    """
     none = CE_None
     debug = CE_Debug
     warning = CE_Warning
     failure = CE_Failure
     fatal = CE_Fatal
```

Comment:
This enum isn't use anywhere, should it be kept here or shall I remove it?

Reply:
Removing it is fine, likely a leftover from porting from Fiona in the early days.